cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst

```diff
@@ -84,111 +101,100 @@ The *pageTitleStream* consumes events in the format returned by the Facebook "Li
+.. include:: _includes/_starting-workflow.txt
 
-    $ curl -w'\n' -v \
-      http://localhost:10000/v3/namespaces/default/apps/WikipediaPipeline/workflows/WikipediaPipelineWorkflow/start
-
-#. From the Standalone CDAP SDK directory, use the Command Line Interface::
-
-    $ cdap-cli.sh start workflow WikipediaPipeline.WikipediaPipelineWorkflow
-  
-    Successfully started flow 'WikipediaPipelineWorkflow' of application 'WikipediaPipeline' with stored runtime arguments '{}'
-
-You can also (optionally) specify these runtime arguments for the *WikipediaPipelineWorkflow*:
+These runtime arguments can be set for the *WikipediaPipelineWorkflow*:
 
 - *min.pages.threshold*: Threshold for the number of pages to exist in the *pageTitleStream* for the workflow to proceed.
   Defaults to 10.
 - *mode*: Set this to 'online' when you wish to download Wikipedia data over the Internet.
-  Defaults to offline, in which the workflow expects Wikipedia data to be in the *wikiStream*.
+  Defaults to 'offline', in which case the workflow expects Wikipedia data to be in the *wikiStream*.
 - *stopwords.file*: The path to the file containing stopwords to filter in the *SparkWikipediaAnalyzer* program.
   If unspecified, no words are considered as stopwords.
 - *vocab.size*: The size of the vocabulary for the *SparkWikipediaAnalyzer* program. Defaults to 1000.
 - *topn.rank*: The number of top words to produce in the *TopNMapReduce* program. Defaults to 10.
 - *num.reduce.tasks*: The number of reduce tasks to set for the *TopNMapReduce* program. Defaults to 1.
 
-Retrieving the Results
-----------------------
-The *WikipediaPipelineApp* contains a CDAP Service *WikipediaService* that can retrieve results from the analysis
-performed by the *WikipediaPipelineWorkflow*.
-
-To start the service:
-
-- Go to the *WikipediaPipelineApp* `application overview page
-  <http://localhost:9999/ns/default/apps/WikipediaPipelineApp/overview/status>`__,
-  click ``WikipediaService`` to get to the service detail page, then click the *Start* button; or
+If you run with the default arguments, you will find that the pipeline starts but then
+stops after the first node, as the number of pages is less than the *min.pages.threshold*:
 
-- From the Standalone CDAP SDK directory, use the Command Line Interface::
+.. image:: _images/wikipedia-data-pipeline-1.png
+   :width: 8in
 
-    $ cdap-cli.sh start service WikipediaPipelineApp.WikipediaService
+Reduce the number of minimum number of pages to zero, and change the mode to *online*, by
+setting either the runtime arguments (which changes the *next* run) or the preferences
+(which changes *all* subsequent runs):
 
-The *WikipediaService* exposes these REST APIs:
+.. image:: _images/wikipedia-data-pipeline-2.png
+   :width: 8in
 
-- Retrieve the list of topics generated by the *SparkWikipediaAnalyzer* program::
+Once the pipeline has run through to the end, you can start the service and query the results.

```

Comment:
for the sake of completeness, should we also add an image that shows a DAG after the workflow has run through to the end?